core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -151,40 +112,34 @@ trait WorkspaceSupport {
                                         attributeSpecs: Option[WorkspaceAttributeSpecs] = None
   ): Future[Workspace] =
     for {
+      _ <- userEnabledCheck
       workspaceContext <- getWorkspaceContext(workspaceName, attributeSpecs)
       _ <- accessCheck(workspaceContext, requiredAction, ignoreLock = false) // throws if user does not have permission
     } yield workspaceContext
 
-  def getWorkspaceContext(workspaceName: WorkspaceName,
-                          attributeSpecs: Option[WorkspaceAttributeSpecs] = None
+  def getWorkspaceContext(
+    workspaceName: WorkspaceName,
+    attributeSpecs: Option[WorkspaceAttributeSpecs] = None
   ): Future[Workspace] =
-    userEnabledCheck.flatMap { _ =>
-      dataSource.inTransaction { dataAccess =>
-        withWorkspaceContext(workspaceName, dataAccess, attributeSpecs)(DBIO.successful)
-      }
-    }
-
-  def withWorkspaceContext[T](workspaceName: WorkspaceName,
-                              dataAccess: DataAccess,
-                              attributeSpecs: Option[WorkspaceAttributeSpecs] = None
-  )(op: (Workspace) => ReadWriteAction[T]) =
-    dataAccess.workspaceQuery.findByName(workspaceName, attributeSpecs) flatMap {
+    workspaceRepository.getWorkspace(workspaceName, attributeSpecs).map {
```

Comment:
Why is the `userEnabledCheck` is not needed?

Reply:
Both call sites of this are methods that were already doing access checks. So I thought it was more appropriate to do the  `userEnabledCheck` in the same location.